examples/step-83/doc/intro.dox

```diff
@@ -0,0 +1,382 @@
+and load the large array, and in the load function re-build the cache
+from the large array. (See also the discussion below on what one
+actually wants to save.)
+
+In the end, serialization has put everything into an "Archive" from
+which one can extract a (sometimes very long) string that one can save
+in bulk into a file. BOOST serialization offers different archives,
+including ones that store the data in ASCII format, in binary format,
+or already compressed with something like gzip to minimize the amount
+of space necessary. The specific type of archive to be used is
+selected at the place where one initiates serialization, and the
+concrete mechanism will become clear when we show the actual
+implementation below.
+
+
+<h3>Checkpointing strategies</h3>
+
+Having discussed the general idea of checkpoint/restart, let us turn
+to some more specific questions one has to answer: (i) What do we
+actually want to save/restore? (ii) How often do we want to write
+checkpoints?
+
+
+<h4>What to save/restore</h4>
+
+We will base this tutorial on step-19, and so let us use it as an
+example in this section. If you think of a program such as step-19,
+then there are a lot of member variables one might want to
+(de)serialize:
+@code
+  template <int dim>
+  class CathodeRaySimulator
+  {
+  public:
+    CathodeRaySimulator();
+
+    void run();
+
+  private:
+    void make_grid();
+    void setup_system();
+    void assemble_system();
+    void solve_field();
+    void refine_grid();
+
+    void create_particles();
+    void move_particles();
+    void track_lost_particle(const typename Particles::ParticleIterator<dim> &        particle,
+                             const typename Triangulation<dim>::active_cell_iterator &cell);
+
+
+    void update_timestep_size();
+    void output_results() const;
+
+    Triangulation<dim>        triangulation;
+    MappingQGeneric<dim>      mapping;
+    FE_Q<dim>                 fe;
+    DoFHandler<dim>           dof_handler;
+    AffineConstraints<double> constraints;
+
+    SparseMatrix<double>      system_matrix;
+    SparsityPattern           sparsity_pattern;
+
+    Vector<double>            solution;
+    Vector<double>            system_rhs;
+
+    Particles::ParticleHandler<dim> particle_handler;
+    types::particle_index           next_unused_particle_id;
+    types::particle_index           n_recently_lost_particles;
+    types::particle_index           n_total_lost_particles;
+    types::particle_index           n_particles_lost_through_anode;
+
+    DiscreteTime time;
+  };
+@endcode
+Do we really need to save all of these to disk? That would presumably
+lead to quite a lot of data that needs to be stored and, if necessary,
+re-loaded.
+
+In practice, one does not save all of this information, but only what
+cannot be reasonably re-computed in different ways. What is saved
+should also depend on also where in the program's algorithm one
```

Comment:
```suggestion
should also depend on where in the program's algorithm one
```

Reply:
Just reading through this step out of interest, hope you don't mind the suggestion :)

Reply:
Quite the contrary! Thank you!